Update list of ports required to build documentation

JDLH opened the pull request:
In the Guide, 7.5. "Updating Documentation", 7.5.1.1 "Preparing Changes", there is a step to install the required ports:
* The list of ports currently reads `libxslt docbook-xsl`. But the `README.md` file in the Guide repo's root has a longer list of ports to install, `libxml2 libxslt docbook-xsl-ns docbook-xml-5.0`.
* Also `docbook-xsl` is an obsolete port, so one can't install it anyway.

This changes the step to install required ports, in 7.5.1.1 "Preparing Changes", to match the list of ports from `README.md`.

As far as I can tell, there is no ticket in Trac documenting this problem with the Guide.

neverpanic commented:
Actually, before merging this, can you make the equivalent change in guide/adoc/project.adoc?

JDLH commented:
Thank you for the review. 

I didn't know about guide/adoc/project.adoc.  Yes, I'll go look for that and make that change. 

And, perhaps I should make two additions to #39: 1. mention guide/adoc/project.adoc. 2. make the appropriate changes in guide/adoc/project.adoc that I made in the xml doc.

JDLH commented:
Hmm. Here's a self-referential problem. 

It is trivial to change the list of ports in the single line of guide/adoc/project.adoc . 

To test if the trival change is correct, it appears I should run either the tool `asciidoc` or the tool `asciidoctor`. There are ports by those same names.  But if testing with these tools is part of responsible authoring of docs, then maybe I should change the list of required ports to include one or both of these.

I think I'm going to defer that to a separate pull request about of better describing how we want to author MacPorts docs in asciidoc (parallel editing in XML and asciidoc? Edit in XML and convert to asciidoc? What about patched `docbookrx` and flawed conversions?).  I'll make a commit with just trivial edit to guide/adoc/project.adoc.

Commit:
Update adoc's list of ports required to build documentation

In the adoc sources of the Guide, guide/adoc/project.adoc,
7.5. "Updating Documentation",  7.5.1.1 "Preparing Changes",
there is a step to install the required ports, to build the
Guide from XML Docbook sources. Update this list of ports
based on README.md in root.

This matches the corresponding change to guide/xml/project.xml
in commit 36ca5d83a2225c415a17d6420b2d9dcf1754a6fe . It is a
minimal-scope one-line change.

This section of the Guide does not describe how to build the
Guide from adoc sources. Adding such a description is out of
scope for this limited patch.

JDLH commented:
Equivalent change made in guide/adoc/project.adoc . 

Tested by running `asciidoctor guide.adoc`. This generates a guide.html document with no errors, which shows this content change in the right place.

I think this branch is ready to merge.

mascguy reviewed (approved):
LGTM. Anyone else?

jmroot commented:
`libxml2` isn't strictly necessary in the install command since it's already a dependency of libxslt.

JDLH commented:
> `libxml2` isn't strictly necessary….

Thank you for the feedback, @jmroot . I suggest not removing `libxml2` in this PR, in the interests of efficiency. The presence of `libxml2` is not harmful.  Its presence makes it consistent with the README.md file in the Guide repo's root. The PR acceptance process is pretty slow already, and adding modification and reapproval cycles to the process will make it slower.

Instead I propose accepting this PR with `libxml2` included, then making a separate action to remove `libxml2` from the README.md file in the Guide repo's root, and in 7.5.1.1 "Preparing Changes" of the Guide content.

In general, I think the documentation process needs to find a way to higher velocity. Keeping changes small, incremental, and numerous will help us get there.

mascguy commented:
> In general, I think the documentation process needs to find a way to higher velocity. Keeping changes small, incremental, and numerous will help us get there.

I strongly concur: It pains me to see documentation-related PRs sitting out here, without any activity, for up to two years! (Which is why I finally added feedback on all of these open PRs, to bring them to folks' attention.)

If these were source code changes in critical areas like MacPorts Base, then that's one thing. But these are documentation-related, and they deserve a more timely response.

jmroot commented:
If some committer wants to volunteer as Chief Documentation Wrangler, that would be great.

mascguy commented:
> If some committer wants to volunteer as Chief Documentation Wrangler, that would be great.

I'm willing to serve as Unofficial Chief Documentation Wrangler. With the caveat that if folks don't respond in a reasonably timely fashion - say, 2-3 weeks, barring any personal tragedies/emergencies - they can expect some serious nagging.  ;-)

mascguy commented:
> > If some committer wants to volunteer as Chief Documentation Wrangler, that would be great.
> 
> I'm willing to serve as Unofficial Chief Documentation Wrangler. With the caveat that if folks don't respond in a reasonably timely fashion - say, 2-3 weeks, barring any personal tragedies/emergencies - they can expect some serious nagging. ;-)

Jokes aside, yes, it would be great to move these along more quickly. And I'm happy to spearhead the effort!
